### PR TITLE
EOS-27356: hare pre-merge CICD job is failing

### PR DESCRIPTION
**Pull request by pavankrishnat**
**Problem:**
Pre-merge tests are failing due to the script used to prepare dependencies
still points to CentOS 7.8 refs and VM used for pre-merge job is running 
on OS CentOS 7.8.
[Jenkins Hare-PreMerge-Test](http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/Hare-PreMerge-Test/)

**Solution:**
1. Replace current VM with centos 7.9 instead with Specs - 
**centos 7.9 , 16 CPU, 16GB RAM, 4 disks/50GB.**
2. Updated the dependencies info in prepare-yum-repos file by correcting 
the centos 7.9 and also verify/Updated all scripts in [Jenkins folder](https://github.com/Seagate/cortx-hare/blob/main/jenkins/) again
as pre-merge job use most of them.

Signed-off-by: pavankrishnat <pavan.k.thunuguntla@seagate.com>

### CODE REVIEW

**Comment by suvratjoshi** [comment]
@pavankrishnat 
PR description and commit message : Please use 74 characters per line.

@pavankrishnat , @gauravchaudhari02 
Can we try running this on our 7.9 VM to confirm if changes work?

**Comment by pavankrishnat** [comment]
@shailesh-vaidya 

Suggest me  which one to add for lustre and motr-uploads repos in `jenkins/prepare-yum-repos` file.
```
centos-7.9.2009-2.0.0-k8/
centos-7.9.2009-2.0.0-latest/
centos-7.9.2009-custom/ 
```
http://cortx-storage.colo.seagate.com/releases/cortx/third-party-deps/centos/

File: jenkins/prepare-yum-repos [hunk 1]
```diff
@@ -27,23 +27,23 @@ STORAGE='http://cortx-storage.colo.seagate.com'
...
+baseurl=$STORAGE/releases/cortx/github/main/centos-7.9.2009/last_successful/
 gpgcheck=0
 name=motr-dev
 enabled=1
 EOF
 
 cat <<EOF > /etc/yum.repos.d/motr_uploads.repo
 [motr-uploads]
-baseurl=$STORAGE/releases/cortx/third-party-deps/centos/centos-7.8.2003-2.0.0-latest/motr_uploads/
+baseurl=$STORAGE/releases/cortx/third-party-deps/centos/centos-7.9.2009-2.0.0-latest/motr_uploads/
```
Comment (pavankrishnat) [inline jenkins/prepare-yum-repos:38]: @shailesh-vaidya 

Suggest me  which one to add for lustre and motr-uploads repos in `jenkins/prepare-yum-repos` file.
```
centos-7.9.2009-2.0.0-k8/
centos-7.9.2009-2.0.0-latest/
centos-7.9.2009-custom/ 
```
http://cortx-storage.colo.seagate.com/releases/cortx/third-party-deps/centos/

**Comment by mssawant** [comment]
@pavankrishnat, please update the commit message as follows,
```
EOS-27356: hare pre-merge CICD job is failing

<please describe the problem, why premerge tests were failing?>

Solution:
- Replace current VM with centos 7.9 instead of centos 7.8 with Specs -
16 CPU, 16GB RAM, 4 disks/50GB.

- Updated the dependencies info in prepare-yum-repos file by correcting
the centos 7.9 and also verify/Updated all scripts in Jenkins folder again
as pre-merge job use most of them.

Signed-off-by: pavankrishnat <pavan.k.thunuguntla@seagate.com>
```

**Review by gauravchaudhari02 — APPROVED** [review]

**Comment by pavankrishnat** [comment]
> @pavankrishnat, please update the commit message as follows,
> 
> ```
> EOS-27356: hare pre-merge CICD job is failing
> 
> <please describe the problem, why premerge tests were failing?>
> 
> Solution:
> - Replace current VM with centos 7.9 instead of centos 7.8 with Specs -
> 16 CPU, 16GB RAM, 4 disks/50GB.
> 
> - Updated the dependencies info in prepare-yum-repos file by correcting
> the centos 7.9 and also verify/Updated all scripts in Jenkins folder again
> as pre-merge job use most of them.
> 
> Signed-off-by: pavankrishnat <pavan.k.thunuguntla@seagate.com>
> ```

Updated

**Comment by mssawant** [comment]
@pavankrishnat, I see pre-merge tests are failing on this PR as well,
```
06:58:53  Traceback (most recent call last):
06:58:53    File "./cfgen", line 23, in <module>
06:58:53      from hax.log import create_logger_directory
06:58:53  ModuleNotFoundError: No module named 'hax'
06:58:53  make[1]: *** [test-cfgen] Error 1
06:58:53  make: *** [test-cfgen] Error 2
06:58:53  Failed command cortx-vm-name#6 with status 2: 
06:58:53                          cd "cortx-hare"
06:58:53                          export PATH=/opt/seagate/cortx/hare/bin:$PATH
06:58:53                          make check
06:58:53                          make test
```

**Comment by pavankrishnat** [comment]
> @pavankrishnat, I see pre-merge tests are failing on this PR as well,
> 
> ```
> 06:58:53  Traceback (most recent call last):
> 06:58:53    File "./cfgen", line 23, in <module>
> 06:58:53      from hax.log import create_logger_directory
> 06:58:53  ModuleNotFoundError: No module named 'hax'
> 06:58:53  make[1]: *** [test-cfgen] Error 1
> 06:58:53  make: *** [test-cfgen] Error 2
> 06:58:53  Failed command cortx-vm-name#6 with status 2: 
> 06:58:53                          cd "cortx-hare"
> 06:58:53                          export PATH=/opt/seagate/cortx/hare/bin:$PATH
> 06:58:53                          make check
> 06:58:53                          make test
> ```

Yes @mssawant those are belongs to unit test. I am working on fixing those [EOS-26855](https://jts.seagate.com/browse/EOS-26855)

**Comment by suvratjoshi** [comment]
> 
> Yes @mssawant those are belongs to unit test. I am working on fixing those [EOS-26855](https://jts.seagate.com/browse/EOS-26855)

@pavankrishnat ,
As discussed can you please try by commenting those test calls for now?
If it passes then may be we can think of commenting those for now and get done with this PR, so that pre-merge will start working.
Later once we are done with EOS-26855, we will enable those tests again for pre-merge as well.

**Comment by pavankrishnat** [comment]
> > Yes @mssawant those are belongs to unit test. I am working on fixing those [EOS-26855](https://jts.seagate.com/browse/EOS-26855)
> 
> @pavankrishnat , As discussed can you please try by commenting those test calls for now? If it passes then may be we can think of commenting those for now and get done with this PR, so that pre-merge will start working. Later once we are done with EOS-26855, we will enable those tests again for pre-merge as well.

Yes @suvratjoshi we (Me and @gauravchaudhari02) tried commenting unit test and ran but it failed with different error Hare-[PreMerge-Test#1050](http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/Hare-PreMerge-Test/1050/console)

**Comment by suvratjoshi** [comment]
@pavankrishnat 
Checked your job :
```
16:57:31  jenkins/test-boot1: line 28: 114022 Segmentation fault      m0crate -S _m0crate-io.yaml
16:57:31  Failed command cortx-vm-name#7 with status 139: 
16:57:31                          cd "cortx-hare"
16:57:31                          jenkins/test-boot1
```
Segmentation fault while doing IO in the test-boot1 file.
It used m0crate command, right?
Currently m0crate is not working, is that correct?
Can we confirm this with motr?

**Comment by pavankrishnat** [comment]
> @pavankrishnat Checked your job :
> 
> ```
> 16:57:31  jenkins/test-boot1: line 28: 114022 Segmentation fault      m0crate -S _m0crate-io.yaml
> 16:57:31  Failed command cortx-vm-name#7 with status 139: 
> 16:57:31                          cd "cortx-hare"
> 16:57:31                          jenkins/test-boot1
> ```
> 
> Segmentation fault while doing IO in the test-boot1 file. It uses m0crate command. Currently m0crate is not working, is that correct? Can we confirm this with motr? Lets raise bug. If it is so, then can we use some other working method for IO here?

@suvratjoshi, for m0crate-io-conf issues @supriyachavan4398 has created [PR#1956](https://github.com/Seagate/cortx-hare/pull/1956)

**Comment by pavankrishnat** [comment]
> > @suvratjoshi, for m0crate-io-conf issues @supriyachavan4398 has created [PR#1956](https://github.com/Seagate/cortx-
> > hare/pull/1956)
> 
> Hi @pavankrishnat , Till we are done with PR #1956 , can we comment this IO section for testing and test pre-merge job just to ensure that there are no other failures?
Yes, @suvratjoshi
@gauravchaudhari02, I Hope we are good to Merge [PR#815](https://github.com/Seagate/cortx-re/pull/815). So that we can Test this PR and close it.

**Comment by pavankrishnat** [comment]
Yes, @suvratjoshi
@gauravchaudhari02, I Hope we are good to Merge PR#815. So that we can Test this PR and close it.

**Comment by gauravchaudhari02** [comment]
> Yes, @suvratjoshi @gauravchaudhari02, I Hope we are good to Merge [PR#815](https://github.com/Seagate/cortx-re/pull/815). So that we can Test this PR and close it.

@pavankrishnat, It is merged now.

**Review by vaibhavparatwar — APPROVED** [review]

**Comment by pavankrishnat** [comment]
@suvratjoshi I ran Premerge job by commenting IO section for testing.
[Hare-PreMerge-Test#1098](http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/Hare-PreMerge-Test/1098/console)

File: jenkins/test-boot1 [hunk 2]
```diff
@@ -26,7 +26,11 @@ HARE_DIR=/opt/seagate/cortx/hare
 export PATH="$HARE_DIR/bin:$PATH"
 
 do_io() {
-    $HARE_DIR/libexec/m0crate-io-conf > _m0crate-io.yaml
+    if [ "$( echo $( grep 'transport_type' $confstore ) | awk -F": " '{print $2}' )" = "lnet" ]; then
```
Comment (pavankrishnat) [inline jenkins/test-boot1:29]: @mssawant @suvratjoshi 
Added Changes as per [PR#1956](https://github.com/Seagate/cortx-hare/pull/1956)

Comment (suvratjoshi) [reply jenkins/test-boot1:29]: Looks like, premerge for this PR is still failing

**Comment by pavankrishnat** [comment]
retest this please

**Comment by pavankrishnat** [comment]
rerun premerge-test

**Comment by pavankrishnat** [comment]
rerun premerge-test

**Comment by mssawant** [comment]
Pre-merge tests are passing now.

**Review by mssawant — APPROVED** [review]

**Pull request merged**